src/libcore/ffi.rs

```diff
@@ -268,35 +268,86 @@ mod sealed_trait {
                reason = "the `c_variadic` feature has not been properly tested on \
                          all supported platforms",
                issue = "44930")]
-    pub trait VaArgSafe {}
+    pub trait VaArgSafe {
+        #[doc(hidden)]
+        #[cfg(not(bootstrap))]
+        unsafe fn __va_arg(ap: &mut super::VaListImpl<'_>) -> Self;
+    }
 }
 
-macro_rules! impl_va_arg_safe {
+macro_rules! impl_va_arg_safe_integer {
     ($($t:ty),+) => {
         $(
             #[unstable(feature = "c_variadic",
                        reason = "the `c_variadic` feature has not been properly tested on \
                                  all supported platforms",
                        issue = "44930")]
-            impl sealed_trait::VaArgSafe for $t {}
+            impl sealed_trait::VaArgSafe for $t {
+                #[cfg(not(all(target_arch = "aarch64", not(target_os = "ios"), not(windows))))]
+                #[cfg(not(bootstrap))]
+                unsafe fn __va_arg(ap: &mut VaListImpl<'_>) -> Self {
+                    va_arg(ap)
+                }
+                #[cfg(all(target_arch = "aarch64", not(target_os = "ios"), not(windows)))]
```

Comment:
You can put the `#[cfg]` on the expression inside the body, I think?

Reply:
Ah yes, I think this is just left over from my first iteration where they had different signatures.